nifi-nar-bundles/nifi-compress-bundle/nifi-compress-processors/src/main/java/org/apache/nifi/processors/compress/ModifyCompression.java

```diff
@@ -0,0 +1,420 @@
+                session.transfer(flowFile, REL_FAILURE);
+                return;
+            }
+        }
+
+        String compressionFormatValue = context.getProperty(OUTPUT_COMPRESSION).getValue();
+        if (compressionFormatValue.equals(CompressionInfo.COMPRESSION_FORMAT_ATTRIBUTE.getValue())) {
+            final String mimeType = flowFile.getAttribute(CoreAttributes.MIME_TYPE.key());
+            if (mimeType == null) {
+                logger.error("No {} attribute exists for {}; routing to failure", CoreAttributes.MIME_TYPE.key(), flowFile);
+                session.transfer(flowFile, REL_FAILURE);
+                return;
+            }
+
+            compressionFormatValue = compressionFormatMimeTypeMap.get(mimeType);
+            if (compressionFormatValue == null) {
+                logger.info("MIME Type of {} is '{}', which does not indicate a supported Compression Format; routing to success without decompressing", flowFile, mimeType);
+                session.transfer(flowFile, REL_SUCCESS);
+                return;
+            }
+        }
+
+        final String decompressionFormat = decompressionFormatValue;
+        final String compressionFormat = compressionFormatValue;
+        final AtomicReference<String> mimeTypeRef = new AtomicReference<>(null);
+        final StopWatch stopWatch = new StopWatch(true);
+
+        String inputFileExtension;
+        String outputFileExtension;
+        try {
+            inputFileExtension = CompressionInfo.fromAllowableValue(decompressionFormat).getFileExtension();
+        } catch (IllegalArgumentException iae) {
+            inputFileExtension = "";
+        }
+        try {
+            outputFileExtension = CompressionInfo.fromAllowableValue(compressionFormat).getFileExtension();
+        } catch (IllegalArgumentException iae) {
+            outputFileExtension = "";
+        }
+
+        try {
+            flowFile = session.write(flowFile, (rawIn, rawOut) -> {
+                final OutputStream compressionOut;
+                final InputStream compressionIn;
+
+                final OutputStream bufferedOut = new BufferedOutputStream(rawOut, STREAM_BUFFER_SIZE);
+                final InputStream bufferedIn = new BufferedInputStream(rawIn, STREAM_BUFFER_SIZE);
+
+                try {
+                    // Decompress data by creating an InputStream for the decompression format, then recompress using the compression format
+                    compressionIn = getCompressionInputStream(decompressionFormat, bufferedIn);
+
+                    int compressionLevel = context.getProperty(COMPRESSION_LEVEL).asInteger();
+                    compressionOut = getCompressionOutputStream(compressionFormat, compressionLevel, mimeTypeRef, bufferedOut);
+
+                } catch (final Exception e) {
+                    closeQuietly(bufferedOut);
+                    throw new IOException(e);
```

Comment:
A message should be added to the exception.
```suggestion
                    throw new IOException(String.format("Input Type [%s] Output Type [%s] Compression operation failed", decompressionFormat, compressionFormat), e);
```